cmd/cql-minerd/integration_test.go

```diff
@@ -409,76 +416,69 @@ func benchDB(b *testing.B, db *sql.DB, createDB bool) {
 		prepareBenchTable(db)
 	}
 
-	var i int32
-	var insertedCount int
-
-	rand.Seed(time.Now().UnixNano())
-	start := (rand.Int31() % 100) * 10000
-
-	//b.Run("benchmark Single INSERT", func(b *testing.B) {
-	//	b.ResetTimer()
-	//	insertedCount = b.N
-	//	for i := 0; i < b.N; i++ {
-	//		_, err = db.Exec("INSERT INTO test ( indexedColumn, nonIndexedColumn ) VALUES"+
-	//			"(?, ?)", int(start)+i, i,
-	//		)
-	//		if err != nil {
-	//			b.Fatal(err)
-	//		}
-	//	}
-	//})
-	//
-	//if createDB {
-	//	prepareBenchTable(db)
-	//}
-	//
-	b.Run("benchmark Multi INSERT", func(b *testing.B) {
+	cleanBenchTable(db)
+
+	var i int64
+	i = -1
+
+	b.Run("benchmark INSERT", func(b *testing.B) {
 		b.ResetTimer()
-		insertedCount = b.N
 		b.RunParallel(func(pb *testing.PB) {
 			for pb.Next() {
-				ii := atomic.AddInt32(&i, 1)
-				_, err = db.Exec("INSERT INTO test ( indexedColumn, nonIndexedColumn ) VALUES"+
-					"(?, ?)", start+ii, ii,
+				ii := atomic.AddInt64(&i, 1)
+				_, err = db.Exec("INSERT INTO insert_table0 ( k, v1 ) VALUES"+
+					"(?, ?)", ROWSTART+ii, ii,
 				)
+				for err != nil && err.Error() == sqlite3.ErrBusy.Error() {
+					// retry forever
+					log.Warnf("ROWSTART+ii = %d retried", ROWSTART+ii)
+					_, err = db.Exec("INSERT INTO insert_table0 ( k, v1 ) VALUES"+
+						"(?, ?)", ROWSTART+ii, ii,
+					)
+				}
 				if err != nil {
 					b.Fatal(err)
 				}
 			}
 		})
 	})
 
-	rowCount := db.QueryRow("SELECT COUNT(1) FROM test")
-	var count int
+	rowCount := db.QueryRow("SELECT COUNT(1) FROM insert_table0")
+	var count int64
 	err = rowCount.Scan(&count)
 	if err != nil {
 		b.Fatal(err)
 	}
-	log.Warnf("Row Count: %d", count)
+	log.Warnf("Row Count: %v", count)
 
 	b.Run("benchmark SELECT", func(b *testing.B) {
 		b.ResetTimer()
 		b.RunParallel(func(pb *testing.PB) {
 			for pb.Next() {
-				i := atomic.AddInt32(&i, 1)
-				index := int(i)%insertedCount + int(start) + 1
-				row := db.QueryRow("SELECT nonIndexedColumn FROM test WHERE indexedColumn = ? LIMIT 1", index)
-				var result int
+				var index int64
+				if createDB { //only data by insert
+					index = rand.Int63n(count-1) + ROWSTART
```

Comment:
gofmt this

Reply:
gofmt did this... Maybe some complex format rules.